### DISCUSSION TITLE
ASR spelling task

### DISCUSSION THREAD

**Discussion by itzsimpl** [NVIDIA-NeMo/NeMo#4231] [Q&A] [answered]
Let's assume we're developing an ASR with which we'd like to allow the user to spell out words when the ASR does not recognise them. For example one could say /K IY2 N OW1 AH0/ for "quinoa", or spell out individual letters /K Y UW1/, /Y UW1/, /AY1/, /EH1 N/, /OW1/, /EY1/ for "q u i n o a". Which model would be most suitable for something like this? Has anyone attempted such a thing? What is the accuracy? Intuitively it seems that pronouncing individual letters in English provides enough data for them to be recognised as such, how about other languages where spelling out letters does not provide this much data and where some individual letters sound very much alike? Will be happy of any discussion on this topic; experiences, models, papers, ...

**Comment by titu1994** [comment, marked as answer]
You can try to use QuartzNet / or Citrinet char / Conformer char models as audio to phoneme output. We have tried once to do it, but the user would speak normally (without accentuating the phoneme specifically), and the model would predict phones. 

There wasn't a standard benchmark to compare against, but in terms of PER, we got it down to <2%. .

For your case specifically, where user explicitly utters the phoneme, we have never tried that. It may work just fine but we have no guarantee it will work.

**Reply by titu1994** [reply]
You could potentially just use a small RNNT model, and in the ground truth text split the text with space for every single character. I say RNNT cause CTC will not be able to work with 4 or 8x striding with such char space tokenization - it would break the T > U assumption required by the loss.

**Reply by titu1994** [reply]
Interestingly, our correctly trained models do quite well on AN4, which while tiny, actually does have people which explicitly spell out the characters, and the model does very well there.
